### PR TITLE
chore(meta): Document Embedlivesample identifier

### PR DESCRIPTION
**Pull request by bsmth**
The `{{Embedlivesample}}` macro currently has two ways of specifying which code blocks to combine into a "result" iframe, and only one of these ways is documented. This PR adds documentation for the "string identifier" method or marking fenced code blocks (HTML/CSS/JS) with an identifier in the info string using:

- ` ```{language} live-sample___{identifier} `

For example, in Markdown, this is used as follows:

````md
## Examples

### Styling a paragraph

In this example, we're using CSS to style paragraphs.

```html live-sample___paragraph-styling
<p>I'm not fancy.</p>
<p class="fancy">But I am!</p>
```

```html
<p>I'm another HTML block. I won't be included in the result.</p>
```

```css live-sample___paragraph-styling
p.fancy {
  color: red;
}
```

Only the `<p>` element with `class="fancy"` will get styled `red`:

{{EmbedLiveSample("paragraph-styling")}}
````

### CODE REVIEW

**Comment by dipikabh** [comment]
Thanks for opening this PR, @bsmth 🙏! I'll take a look now.

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 1]
```diff
@@ -76,12 +155,60 @@ In many cases, you may be able to add the `EmbedLiveSample` or `LiveSampleLink`
...
+- HTML
+- CSS
+- JavaScript
+- Result
+
+Write the code blocks in the respective subsections listed above.
+
+In the **Result** subsection, add the call to the `EmbedLiveSample` macro. Preferably, include some more prose in this subsection to describe the result.
+
+If you're not using a particular language type (for example, if you are not using JavaScript) or if you are hiding it, then you should omit the corresponding heading.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:171]:
```suggestion
If you're not using a particular language type (for example, if you are not using JavaScript) or if you are hiding the code block, then you should omit the corresponding heading.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 2]
```diff
@@ -76,12 +155,60 @@ In many cases, you may be able to add the `EmbedLiveSample` or `LiveSampleLink`
...
+
+### Hidden code
+
+Sometimes you just want to display the static code block that is pertinent to the example rendered within a page. However you still need HTML, CSS and JavaScript to render such an example.
+
+To achieve this you can hide any code blocks that are not relevant with the `hidden` class. If you do this, omit the `### HTML/CSS/JavaScript` headings for the hidden code blocks.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:177]:
```suggestion
### Hiding code

Sometimes you just want to display the static code block pertinent to the example rendered within a page. However, you still need the HTML, CSS, and JavaScript code blocks to render such an example.

To achieve this, you can hide any code blocks that are not relevant by using the `hidden` class. If you do this, omit the `### HTML/CSS/JavaScript` headings for the hidden code blocks.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 3]
```diff
@@ -76,12 +155,60 @@ In many cases, you may be able to add the `EmbedLiveSample` or `LiveSampleLink`
...
+#### Result
+
+Only the `<p>` element with `class="fancy"` will get styled `red`.
+
+\{{EmbedLiveSample("Styling a paragraph")}}
+````
+
+### Turning snippets into live samples
+
+One common use case is to take existing code snippets already shown on MDN and turn them into live samples.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:211]:
```suggestion
One common use case is to take existing code snippets already shown on MDN and turning them into live samples.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 4]
```diff
@@ -140,15 +267,57 @@ el.onclick = function () {
...
+
+{{EmbedLiveSample('grouping_code_blocks_by_heading')}}
+
+Here is a link that results from calling these code blocks via `\{{LiveSampleLink('grouping_code_blocks_by_heading', 'Live sample demo link')}}`:
+
+{{LiveSampleLink('grouping_code_blocks_by_heading', 'Live sample demo link')}}
+
+### Grouping code blocks by identifier
+
+This HTML creates a paragraph and some blocks to help us position and style a message.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:282]:
```suggestion
This HTML creates a paragraph and some blocks to help us position and style a message. The `live-sample___hello-world` class has been added to the `html` language identifier for this code block.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 5]
```diff
@@ -140,15 +267,57 @@ el.onclick = function () {
...
+
+```html live-sample___hello-world
+<p>A simple example of the live sample system in action.</p>
+<div class="box">
+  <div id="item">Hello world! Welcome to MDN</div>
+</div>
+```
+
+The CSS code styles the box as well as the text inside it.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:291]:
```suggestion
The CSS code styles the box as well as the text inside it. The `live-sample___hello-world` class has been added to the `css` language identifier for this code block.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 6]
```diff
@@ -140,15 +267,57 @@ el.onclick = function () {
...
+#item {
+  font-weight: bold;
+  text-align: center;
+  font-family: sans-serif;
+  font-size: 1.5em;
+}
+```
 
-Here is the result of running the code blocks above via `\{{EmbedLiveSample('Live_sample_demo')}}`:
+This code attaches an event handler to the "Hello world!" text that makes an alert appear when it is clicked.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:309]:
```suggestion
This JavaScript code attaches an event handler to the "Hello world!" text that makes an alert appear when it is clicked. The `live-sample___hello-world` class has been added to the `js` language identifier for this code block as well.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 7]
```diff
@@ -140,15 +267,57 @@ el.onclick = function () {
...
-{{EmbedLiveSample('Live_sample_demo')}}
+```js live-sample___hello-world
+const el = document.getElementById("item");
+el.onclick = function () {
+  alert("Owww, stop poking me!");
+};
+```
 
-Here is a link that results from calling these code blocks via `\{{LiveSampleLink('Live_sample_demo', 'Live sample demo link')}}`:
+The result of running the code blocks above via `\{{EmbedLiveSample('hello-world')}}`:
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:318]:
```suggestion
We get this output by running the code blocks above using the string identifier `hello-world` in the `\{{EmbedLiveSample('hello-world')}}` macro call:
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks, @bsmth. This is good overall. I've added some edit suggestions.

**Comment by bsmth** [comment]
Thank you for the comments, @dipikabh! 🙏🏻

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 8]
```diff
@@ -6,31 +6,109 @@ page-type: mdn-writing-guide
 
 {{MDNSidebar}}
 
-MDN supports turning sample code displayed in articles into running samples the reader can look at in action. These are called _live samples_ and allows users to see what code output looks like. It also makes documentation more dynamic and instructive.
+MDN supports turning code blocks in articles into running samples the reader can see in action. These are called _live samples_ and allows users to see what code output looks like. It also makes documentation dynamic and instructive.
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:9]: Yes, this is great, tnx.

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 9]
```diff
@@ -6,31 +6,110 @@ page-type: mdn-writing-guide
...
+<!-- HTML code -->
+```
+
+```css
+/* CSS code */
+```
+
+\{{EmbedLiveSample("Examples")}}
+````
+The live sample system allows grouping of code blocks in different ways to effectively display the output. The next section describes these methods.
```
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:38]:
```suggestion
The live sample system allows grouping of code blocks in different ways to effectively display the output. The next section describes these methods.

```

**Comment by bsmth** [comment]
Thanks @dipikabh - let's take a look again when you're back.

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 10]
```diff
@@ -76,12 +158,60 @@ In many cases, you may be able to add the `EmbedLiveSample` or `LiveSampleLink`
...
+In the **Result** subsection, add the call to the `EmbedLiveSample` macro. Preferably, include some more prose in this subsection to describe the result.
+
+If you're not using a particular language type (for example, if you are not using JavaScript) or if you are hiding the code block, then you should omit the corresponding heading.
+
+### Hiding code
+
+Sometimes you just want to display the static code block pertinent to the example rendered within a page. However, you still need the HTML, CSS, and JavaScript code blocks to render such an example.
 
-One common use case is to take existing code snippets already shown on MDN and turn them into live samples.
+To achieve this, you can hide any code blocks that are not relevant by using the `hidden` class. If you do this, omit the `### HTML/CSS/JavaScript` headings for the hidden code blocks.
```
Comment (pepelsbey) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]: It’s probably not the right occasion to talk about this, but I just noticed that `css` or `html` bits that we add to fenced code blocks are called “classes”, which might be confusing. They’re not HTML classes, they’re rather [language IDs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting). As for the `hidden` bit, it’s not even a language ID, it’s something custom that we use.

We might want to separate those in the docs to clarify what’s language ID and what’s a custom command, attribute, or tag. Let me know if it makes sense, and I’ll start a discussion in some other, more proper place :)

Comment (bsmth) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]: No that's a good clarification, language ID is good. I've also seen 'info string' which covers all cases, maybe we could use that instead?

Comment (pepelsbey) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]: Let’s maybe move this discussion out of the issue to some team call or a chat? :)

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 11]
```diff
@@ -6,31 +6,112 @@ page-type: mdn-writing-guide
...
 ## How does the live sample system work?
 
-The live sample system gathers up all the code in a group, merges it into one HTML file, and then renders that HTML in an {{HTMLElement("iframe")}}. A live sample therefore consists of two pieces:
+The live sample system groups code blocks, merges them into HTML and renders the HTML in an {{HTMLElement("iframe")}}.
+A live sample consists of two parts:
+
+- One or more code blocks grouped together
+- A macro call that shows the result of the combined code blocks in an {{HTMLElement("iframe")}}
+
+Each [code block](/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#example_code_blocks) containing code for the output has a language identifier — `html`, `css`, or `js` — that specifies whether it's HTML, CSS, or JavaScript code. The language classes must be on the corresponding blocks of code, and a macro call (`EmbedLiveSample`) must be present in the page to display the output:
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:23]:
```suggestion
Each [code block](/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#example_code_blocks) containing code for the output has a language identifier — `html`, `css`, or `js` — that specifies whether it's HTML, CSS, or JavaScript code. The language identifiers must be on the corresponding blocks of code, and a macro call (`EmbedLiveSample`) must be present in the page to display the output:
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md (hunk 10 above)
Comment (dipikabh) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]: That is an interesting point @pepelsbey and would be good to clarify in our docs and to use consistently. Thanks for spotting!

We use language ID and like @bsmth mentioned "info strings" in our guidelines for [Example code blocks](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#example_code_blocks) and further below on the same page in the section [Additional classes (info strings)](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#additional_classes_info_strings). The second section has a link to these [GitHub docs](https://github.github.com/gfm/#info-string):

> The first word of the [info string](https://github.github.com/gfm/#info-string) is typically used to specify the language of the code sample, and rendered in the class attribute of the code tag.

Perhaps the section title "Additional classes (info strings)" itself should be updated to "Additional info strings"(outside the scope of this PR).

We can update a few instances (class vs language ID) on the pages touched via this PR.

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 12]
```diff
@@ -140,15 +270,57 @@ el.onclick = function () {
...
+
+Here is a link that results from calling these code blocks via `\{{LiveSampleLink('grouping_code_blocks_by_heading', 'Live sample demo link')}}`:
 
-Here is the result of running the code blocks above via `\{{EmbedLiveSample('Live_sample_demo')}}`:
+{{LiveSampleLink('grouping_code_blocks_by_heading', 'Live sample demo link')}}
 
-{{EmbedLiveSample('Live_sample_demo')}}
+### Grouping code blocks by identifier
+
+This HTML creates a paragraph and some blocks to help us position and style a message. The `live-sample___hello-world` class has been added to the `html` language identifier for this code block.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:285]:
```suggestion
This HTML creates a paragraph and some blocks to help us position and style a message. The `live-sample___hello-world` string has been added to the `html` language identifier for this code block.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 13]
```diff
@@ -140,15 +270,57 @@ el.onclick = function () {
...
+
+```html live-sample___hello-world
+<p>A simple example of the live sample system in action.</p>
+<div class="box">
+  <div id="item">Hello world! Welcome to MDN</div>
+</div>
+```
+
+The CSS code styles the box as well as the text inside it. The `live-sample___hello-world` class has been added to the `css` language identifier for this code block.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:294]:
```suggestion
The CSS code styles the box as well as the text inside it. The `live-sample___hello-world` string has been added to the `css` language identifier for this code block.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 14]
```diff
@@ -140,15 +270,57 @@ el.onclick = function () {
...
+
+#item {
+  font-weight: bold;
+  text-align: center;
+  font-family: sans-serif;
+  font-size: 1.5em;
+}
+```
+
+This JavaScript code attaches an event handler to the "Hello world!" text that makes an alert appear when it is clicked. The `live-sample___hello-world` class has been added to the `js` language identifier for this code block as well.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:312]:
```suggestion
This JavaScript code attaches an event handler to the "Hello world!" text that makes an alert appear when it is clicked. The `live-sample___hello-world` string has been added to the `js` language identifier for this code block as well.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md [hunk 15]
```diff
@@ -6,31 +6,112 @@ page-type: mdn-writing-guide
...
-- If the ID belongs to a heading, the group includes all the code blocks that are after that heading and before the next heading of the same heading level. Note that code blocks under subheadings of the specified heading are all used; if this is not the effect you want, use an ID on a block element instead.
+- If the ID belongs to a heading, the group includes all the code blocks that are after that heading and before the next heading of the same heading level. Note that code blocks under subheadings of the specified heading are all used; if this is not the effect you want, use an ID on a block element or use a string identifier instead.
+
+To group code blocks using a string identifier, add a class that begins with `live-sample___` to the code block's language identifier. Add a unique identifier to the class prefix in the format `live-sample___{IDENTIFIER}`. The `live-sample___` prefix along with the identifier helps the live sample system recognize and group code blocks correctly. For example, `live-sample___color-picker` uses `color-picker` as the identifier following the `live-sample___` prefix.
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:83]:
```suggestion
To group code blocks using a string identifier, add an info string that begins with `live-sample___` to the code block's language identifier. Add a unique identifier to the prefix in the format `live-sample___{IDENTIFIER}`. The `live-sample___` prefix along with the identifier helps the live sample system recognize and group code blocks correctly. For example, `live-sample___color-picker` uses `color-picker` as the identifier following the `live-sample___` prefix.
```

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md (hunk 10 above)
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]:
```suggestion
To achieve this, you can hide any code blocks that are not relevant by adding the `hidden` info string to the language identifier. If you do this, omit the `### HTML/CSS/JavaScript` headings for the hidden code blocks.
```

**Review by dipikabh — APPROVED** [review]
This is looking good, @bsmth. Thanks for the updates.

 There's a few suggestions following Vadim's comment about saying "lang ID" vs "class". Leaving a +1 so you can merge after the changes.

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md (hunk 15 above)
Comment (bsmth) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:83]: Thank you! I've modified it slightly, but it's looking better now

File: files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md (hunk 10 above)
Comment (bsmth) [reply files/en-us/mdn/writing_guidelines/page_structures/live_samples/index.md:180]: Thanks both! I would go for "info string" here for now. We can discuss separately later if you like.

* See also in CommonMark: https://spec.commonmark.org/0.28/#info-string

**Review by pepelsbey — APPROVED** [review]

**Comment by bsmth** [comment]
Thanks, both. Going to merge now 🎉

**Pull request merged**